.github/workflows/pr_master.yml

```diff
@@ -11,6 +11,30 @@ env:
   ROCM_VERSION: "6.3.1"
 
 jobs:
+  check_whitespace:
+    name: Check Whitespace
+    runs-on: ubuntu-22.04
+    steps:
+      - run: |
+          fd \
+            --exclude '*.bc' \
+            --exclude '*.exe' \
+            --exclude '*.lib' \
+            --exclude ext/detours \
+            --strip-cwd-prefix \
+            --type file \
+            --exec bash -c '
+              diff \
+                --unified \
+                --label "a/$0" \
+                --label "b/$0" \
+                <(cat "$0") \
+                <(sed --regexp-extended 's/\s+$//; $a\') || status="$?"
+
+              if (($status != 1)); then
+                exit "$(($status + 1))"
+              fi
+            '
```

Comment:
> Is there a way we can add those checks into our CI? Preferably with existing github action. We do have checks for formatting of Rust code, we can add more checks

Commit https://github.com/vosen/ZLUDA/pull/530/commits/536d8092458f974bae9bbc55f330ab3eb6a40cbf should be able to do this. What is your preferred way of installing `fd`?